src/region.js

```diff
@@ -204,12 +208,38 @@ Marionette.Region = Marionette.Object.extend({
     return Backbone.$(el, Marionette._getValue(this.options.parentEl, this));
   },
 
+  // Replace the region's DOM element with the view's DOM element.
+  replaceEl: function(view) {
+    // empty el so we don't save any non-destroyed views
+    this.$el.contents().detach();
```

Comment:
I'm with @samccone and @jridgewell. If you look at [my original PR](https://github.com/marionettejs/backbone.marionette/commit/fe0d47fdc6162484bff2c97918ef2720e05ea4a0#diff-b8da5292cb07ec757ac64e78ff5af5c1R143) I didn't do anything fancy. I just replaced the old `el` with the new `el` so that we retain references to everything but we're not storing anything unnecessary.